src/game/SpellEffects.cpp

```diff
@@ -2850,48 +2857,50 @@ void Spell::EffectTameCreature(SpellEffectIndex /*eff_idx*/)
     if (plr->IsPvP())
         pet->SetPvP(true);
 
-    if (!pet->InitStatsForLevel(creatureTarget->getLevel()))
-    {
-        sLog.outError("Pet::InitStatsForLevel() failed for creature (Entry: %u)!", creatureTarget->GetEntry());
-        delete pet;
-        return;
-    }
+    uint32 pet_number = sObjectMgr.GeneratePetNumber();
+    pet->GetCharmInfo()->SetPetNumber(pet_number, true);
 
-    pet->GetCharmInfo()->SetPetNumber(sObjectMgr.GeneratePetNumber(), true);
-    // this enables pet details window (Shift+P)
-    pet->AIM_Initialize();
-    pet->InitPetCreateSpells();
-    pet->SetHealth(pet->GetMaxHealth());
+    pet->GetCharmInfo()->SetReactState(REACT_DEFENSIVE);
 
-    // "kill" original creature
+    pet->InitStatsForLevel(creatureTarget->getLevel());
+
+    pet->SetHealthPercent(creatureTarget->GetHealthPercent());
+
+    // destroy creature object
     creatureTarget->ForcedDespawn();
 
+    // add pet object to the world
+    pet->GetMap()->Add((Creature*)pet);
+
     // prepare visual effect for levelup
     pet->SetUInt32Value(UNIT_FIELD_LEVEL, creatureTarget->getLevel() - 1);
 
-    // add to world
-    pet->GetMap()->Add((Creature*)pet);
-
     // visual effect for levelup
     pet->SetUInt32Value(UNIT_FIELD_LEVEL, creatureTarget->getLevel());
 
+    // this enables pet details window (Shift+P)
+    pet->AIM_Initialize();
+    pet->InitPetCreateSpells();
+
     // caster have pet now
     plr->SetPet(pet);
 
-    pet->SavePetToDB(PET_SAVE_AS_CURRENT);
     plr->PetSpellInitialize();
+
+    pet->SavePetToDB(PET_SAVE_AS_CURRENT);
 }
 
 void Spell::EffectSummonPet(SpellEffectIndex eff_idx)
 {
     uint32 petentry = m_spellInfo->EffectMiscValue[eff_idx];
 
     Pet* OldSummon = m_caster->GetPet();
+    Pet* NewSummon = new Pet;
 
-    // if pet requested type already exist
-    if (OldSummon)
+    if (m_caster->getClass() == CLASS_HUNTER)
     {
-        if ((petentry == 0 || OldSummon->GetEntry() == petentry) && OldSummon->getPetType() != SUMMON_PET)
+        // if pet type is already roaming the world
+        if (OldSummon)
         {
             // pet in corpse state can't be summoned
             if (OldSummon->isDead())
```

Comment:
you are right it does. Thank you for pointing, and sorry for not understanding at first. I see it very clearly now.